client/my-sites/activity/filterbar/index.jsx

```diff
@@ -52,13 +60,32 @@ export class Filterbar extends Component {
 		this.setState( { showActivityTypes: false } );
 	};
 
+	toggleIssueTypesSelector = () => {
+		this.setState( {
+			showActivityTypes: false,
+			showActivityDates: false,
+			showIssueTypes: ! this.state.showIssueTypes,
+		} );
```

Comment:
This can potentially be using an inconsistent state if multiple updates are batched together (you can read [this](https://reactjs.org/docs/react-component.html#setstate) and particularly [this](https://stackoverflow.com/questions/48563650/does-react-keep-the-order-for-state-updates/48610973#48610973) to understand why). 

To get around this, we use the callback form of `setState()`, and that way we ensure that we're using the up-to-date previous state:

```suggestion
		this.setState( ( prevState ) => ( {
			showActivityTypes: false,
			showActivityDates: false,
			showIssueTypes: ! prevState.showIssueTypes,
		} ) ) ;
```

Reply:
Thanks for your feedback, we will get this fixed. 

1202076982646589-as-1202419589266159